docs/data/data-grid/filtering/customization.md

```diff
@@ -179,6 +179,36 @@ The demo below shows how to anchor the filter panel to the toolbar button instea
 
 {{"demo": "CustomFilterPanelPosition.js", "bg": "inline", "defaultCodeOpen": false}}
 
+### Optimize performance
+
+There is a new set of APIs with a more efficient interface that are going to be used by default at the next major release, V7.
+
+You can use them right now to make your custom filters faster. Instead of receiving a `GridCellParams` argument, they receive the parameters listed below.
+
+```typescript
+const colDef = {
+  /* ...other column props */
+  getApplyFilterFnV7: (filterItem: GridFilterItem) => {
```

Comment:
Since `getApplyFilterFn` is still required in the interface, users will have to define both `getApplyFilterFnV7` and `getApplyFilterFn` if they want to benefit from faster filtering:

<img width="994" src="https://github.com/mui/mui-x/assets/13808724/9d3bb272-ed52-4d35-8725-cfdc2c6dfcf3">

We can make `getApplyFilterFn` optional, with a downside that TS won't tell you that either `getApplyFilterFn` or `getApplyFilterFnV7` is required. What do you think about such a trade-off?

Reply:
I can extract like before:

```ts
type FilterInterface =
 | { getApplyFilterFn: ... }
 | { getApplyFilterFnV7: ... }

type GridFilterOperator =
  FilterInterface & {
    ...
  }
```

Reply:
IMO, if the DX is better with this, then more verbose types are fine 👍

Reply:
I've refactored so that at least one of them must be defined.

Reply:
I've removed this, making `.getApplyFilterFn` optional would break type-check for our users.